### PR TITLE
Add Azure Function App for MailRisk Sentinel Connector

### PR DESCRIPTION
**Pull request by ingin97**
Required items, please complete
   
   Change(s):
   - Added Azure Function App code
   - Update link to Data Connector deploy script

   Reason for Change(s):
   - Need deployed Azure Function App code in the repo

   Version Updated:
   - No, not yet published.

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Comment by v-rucdu** [comment]
@ingin97 : Please act on the Code QL alerts generated.

**Comment by v-rucdu** [comment]
@ingin97 : We tried deploying the azure function but still getting the error **function runtime is unreachable**.

**Suggestions to fix this error:** 
 1. After extracting the zip package, we found that there is a folder named **MailRiskSentinelIntegration** and inside this all the files are included, whereas expected behavior is that the Azure function related files should be at the root, so please delete this folder and add all the content at the root in the zip file. 
       You can refer [**Alibaba Cloud**](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Alibaba%20Cloud/DataConnectors) Azure function zip file to reference.
 3. Since, this is python based azure function we must include the **.python_packages** folder at the root in the zip folder. (Please refer Alibaba Cloud function package file)
 4. Also, we see that the zip package contains some irrelevant files, viz. **SecurePractice_MailRisk.json** and **azuredeploy_SecurePractice_MailRisk.json** file as well. Please exclude them and other irrelevant files from the zip package.
 
Thanks!!

**Comment by ingin97** [comment]
Hi @v-rucdu,

1. My mistake, I have updated the files to be in the root of the zip folder now. 
2. I was not quite sure how to create the **.python_packages** folder of the dependencides properly. The command I used was: 
`python -m pip install -r requirements.txt --target=./.python_packages/lib/site-packages/`. Is this correct?
3. They are removed from the zip.

File: Solutions/MailRisk/Data Connectors/azuredeploy_SecurePractice_MailRisk.json [hunk 1]
```diff
@@ -228,7 +222,7 @@
             "type": "Microsoft.Web/sites",
             "apiVersion": "2022-03-01",
             "name": "[variables('FunctionName')]",
-            "location": "[parameters('location')]",
+            "location": "[variables('location')]",
             "dependsOn": [
                 "[resourceId('Microsoft.Storage/storageAccounts', tolower(variables('FunctionName')))]",
                 "[resourceId('Microsoft.Insights/components', variables('FunctionName'))]",
```
Comment (v-rucdu) [inline Solutions/MailRisk/Data Connectors/azuredeploy_SecurePractice_MailRisk.json:228]: @ingin97 : To fix **functions runtime is unreachable error** please modify below 2 sections at mentioned line numbers.

1. Modify line# 231 from: **"kind": "functionapp", ** To:  **"kind": "functionapp,linux", **
2. Append below section at line@ 240 under properties, 
               **"siteConfig": {
                                  "linuxFxVersion": "python|3.8"
                              }**

File: Solutions/MailRisk/Data Connectors/SecurePractice_MailRisk.json [hunk 2]
```diff
@@ -111,7 +111,7 @@
...
+            "description": "Use this method for automated deployment of the MailRisk data connector using an ARM Template.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinelmailriskazuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n3. Enter the **Workspace ID**, **Workspace Key**, **Secure Practice API Key**, **Secure Practice API Secret** \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**.\n5. Click **Purchase** to deploy."
```
Comment (v-rucdu) [inline Solutions/MailRisk/Data Connectors/SecurePractice_MailRisk.json:114]: To keep the URLs consistent, we have made small change please update it, 

**Old Value:** https://aka.ms/sentinelmailriskazuredeploy
**New Value:** https://aka.ms/sentinel-mailrisk-azuredeploy

Thanks!!

**Review by v-rucdu — CHANGES_REQUESTED** [review]

**Comment by v-rucdu** [comment]
@ingin97 : After making changes to azure deploy file, please test the function app once and share the successful execution screenshot from the Monitor tab as we do not have the credentials to test it End to End.... Thanks!!!

**Comment by ingin97** [comment]
Hi @v-rucdu! Thank you so much, now the deployment of the function app works! 🚀🎉 
image

**NB**: It only worked after I changed the Application Settings for **WORKSPACE_KEY**, **API_KEY**, **API_SECRET** from using the keyvault reference to raw text. Do you know how I can fix the references in the keyvault to return the correct value?

Thanks!!

**Review by v-rucdu — APPROVED** [review]

File: Solutions/MailRisk/Package/mainTemplate.json [hunk 3]
```diff
@@ -196,7 +196,7 @@
...
+                      "description": "Use this method for automated deployment of the MailRisk data connector using an ARM Template.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinelmailriskazuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n3. Enter the **Workspace ID**, **Workspace Key**, **Secure Practice API Key**, **Secure Practice API Secret** \n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**.\n5. Click **Purchase** to deploy.",
```
Comment (v-rucdu) [inline Solutions/MailRisk/Package/mainTemplate.json:199]: @ingin97: Please update the package to have correct short-link https://aka.ms/sentinel-mailrisk-azuredeploy... Thanks!!

Comment (ingin97) [reply Solutions/MailRisk/Package/mainTemplate.json:199]: @v-rucdu : Ah, I forgot to update the package. Here is a new PR: https://github.com/Azure/Azure-Sentinel/pull/7714

**Pull request merged**